tests/dtm/test_multi_m0d_process_restart.py

```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
+        resp = que_1.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info_write = resp[1]
+
+        self.log.info("Step 3a: Start READ Operations in loop in background for data written in "
+                      "step 1:")
+        proc_read_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                               args=(workload_info_write, que_1, False, True,
+                                                     True, self.test_cfg['loop_count']))
+        proc_read_op.start()
+        self.log.info("Step 3b: Start write Operations in loop in background:")
+        args = {'bucket_prefix': self.bucket_name, 'object_prefix': self.object_name,
+                'no_of_clients': self.test_cfg['clients'],
+                'no_of_samples': self.test_cfg['samples'], 'log_file_prefix': log_file_prefix,
+                'queue': que_2, 'loop': self.test_cfg['loop_count'], 'created_bucket': bucket_list}
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op, kwargs=args)
+        proc_write_op.start()
+
+        time.sleep(self.delay)
+
+        self.log.info("Step 4: Perform m0d Process Restart on %s unique data pods during"
+                      " Write and Read Operations",self.kvalue)
+        pods_list = []
+        m0d_proc_list = []
+        count = 0
+        self.log.info("Stop m0d process with 120 seconds delay in between")
+        while count < self.kvalue:
+            pod_selected, container = self.master_node_list[0].select_random_pod_container(
+                const.POD_NAME_PREFIX, const.MOTR_CONTAINER_PREFIX)
+            if pod_selected in pods_list:
+                continue
+            else:
+                pods_list.append(pod_selected)
+                count = count+1
+            proc = multiprocessing.Process(target = self.dtm_obj.process_restart_mp,
+                                           args = (self.master_node_list[0],pod_selected,container,
+                                                   self.m0d_process, 300, m0d_que))
+            proc.start()
+            m0d_proc_list.append(m0d_proc_list)
+            time.sleep(120)
+        self.log.info("Sleep for 300+120 seconds")
+        time.sleep(420)
+        self.log.info("Verify if all the processes recovered")
+        for each in m0d_proc_list:
+            if each.is_alive():
+                each.join()
+            resp = m0d_que.get()
+            assert_utils.assert_true(resp, "Failure during m0d process restart/recovery")
+
+        self.log.info("Step 5: Verify hctl status if all services are online")
+        resp = self.ha_obj.poll_cluster_status(self.master_node_list[0], 300)
+        assert_utils.assert_true(resp[0],resp[1])
+
+        self.log.info("Step 6a: Wait for READ Operation to complete.")
+        if proc_read_op.is_alive():
+            proc_read_op.join()
+        resp = que_1.get()
+        assert_utils.assert_true(resp[0], resp[1])
```

Comment:
I don't see retry count getting mentioned, so aren't we expecting failures at least when m0d gets down?